### ISSUE TITLE
GLView inside react-navigation 5

### ISSUE THREAD

**Issue by trexpert** [expo/expo#8833]
### Summary
If using expo-gl inside react-navigation 5 the app crashs with an undefined is not an object exception (evaluating glexVersion.split) and (evaluating viewport[2]).
Sometimes you need to reload the app 2-3 times in order to reproduce this error.
The error happens on android device only. Been able to reproduce on real device and simulator.

Might be related to https://github.com/expo/expo/issues/7623 but ticket is closed.

### Environment
```
  Expo CLI 3.21.5 environment info:
    System:
      OS: Windows 10 10.0.19041
    Binaries:
      Node: 12.16.3 - C:\Program Files\nodejs\node.EXE
      Yarn: 1.22.4 - C:\Program Files (x86)\Yarn\bin\yarn.CMD
      npm: 6.14.4 - C:\Program Files\nodejs\npm.CMD
    npmPackages:
      expo: ~37.0.3 => 37.0.12
      react: ~16.9.0 => 16.9.0
      react-dom: ~16.9.0 => 16.9.0
      react-native: https://github.com/expo/react-native/archive/sdk-37.0.1.tar.gz => 0.61.4
      react-native-web: ~0.11.7 => 0.11.7
```

### Reproducible Demo
Snack: https://snack.expo.io/@trexpert/rn5-expo-gl

### Steps to Reproduce
```
expo init
yarn add @react-navigation/native @react-navigation/stack
expo install react-native-gesture-handler react-native-reanimated react-native-screens react-native-safe-area-context @react-native-community/masked-view expo-gl
```
Add simple App.js:
```
import React from 'react';
import { View } from 'react-native';
import { GLView } from 'expo-gl'
import { NavigationContainer } from '@react-navigation/native'
import { createStackNavigator } from '@react-navigation/stack'
import { CardStyleInterpolators, TransitionPresets } from '@react-navigation/stack'

const Stack = createStackNavigator()

const Splash = () => {
  console.log('render splash: ', true)

  const onContextCreated = () => { }

  return (
    <View style={ { flex: 1, backgroundColor: 'red' } }>
      <GLView style={ { flex: 1, backgroundColor: 'yellow' } } onContextCreate={ onContextCreated } />
    </View>
  )
}

export default function App () {

  console.log('render app: ', true)

  return (
    <View style={ { flex: 1 } }>
      <NavigationContainer>
        <Stack.Navigator screenOptions={ {
          cardStyleInterpolator: CardStyleInterpolators.forNoAnimation
        } }>
          <Stack.Screen name="Splash" component={ Splash } />
        </Stack.Navigator>
      </NavigationContainer>
    </View>
  )
}
```

### Expected Behavior vs Actual Behavior
It should not crash :-)

if i remove the cardStyleInterpolator within the screenOptions everything runs fine, 

it looks like there are no undefined checks within the GLView.js code (Line ~542):
```
  // Drawing buffer width/height
  // TODO(nikki): Make this dynamic
  const viewport = gl.getParameter(gl.VIEWPORT);
  gl.drawingBufferWidth = viewport[2];
  gl.drawingBufferHeight = viewport[3];
```

Same here (GLView.js - line 507):
```
  // determine the prototype to use, depending on OpenGL ES version
  const glesVersion = gl.getParameter(gl.VERSION);
  const supportsWebGL2 = parseFloat(glesVersion.split(/[^\d.]+/g).join(' ')) >= 3.0;
```

**Comment by trexpert** [comment]
Any news?

**Comment by tsapeta** [comment]
Nope, we're very busy with SDK38 release right now. Anyway, in my opinion there shouldn't be any *undefined checks* there, the `gl` object must be present there and if it isn't then something really weird happened.
After reading the code you provided, one thing that comes to my mind is the way how you render the `Splash` and the fact that you pass a different `onContextCreate` function on every render call. That might be a problem. This function should be static, that means you shouldn't update this prop. Maybe wrap it into `React.useState` or move to the global scope so the reference to `onContextCreate` is always the same?

**Comment by trexpert** [comment]
Unfortunately it doesnt change anything. Moving onContextCreate to a static context (outside the functional component) it produces the same error.

**Comment by danmaas** [comment]
To be clear, the issue isn't that `gl` is undefined, it's that return values from calls like `gl.getParameter(gl.VERSION)` are undefined.

#7623 mentions that this is fixed as of expo-gl-cpp v8.2.0, and @pacoelayudante 's comment includes a work-around for earlier binary releases

However, I am now seeing this same error from users who are running the brand-new iOS 14.0 beta. I wonder if the underlying cause is different, since the fix in expo-gl-cpp v8.2.0 only changes the native Android code.

**Comment by trexpert** [comment]
Problem solved with [EXPO SDK 38](https://blog.expo.io/expo-sdk-38-is-now-available-ab6cd30ca2ee) using expo-gl 8.3.1.
